tests/mcp/test_server_bind.py

```
@@ -0,0 +1,85 @@
+
+pytest.importorskip("mcp", reason="MCP optional dependency not installed")
+
+from athf.mcp.server import DEFAULT_HOST, is_exposed_host, main  # noqa: E402
+
+
+class TestIsExposedHost:
+    @pytest.mark.parametrize("host", ["127.0.0.1", "localhost", "::1", "[::1]", "LOCALHOST", " 127.0.0.1 "])
+    def test_loopback_is_not_exposed(self, host):
+        assert is_exposed_host(host) is False
+
+    @pytest.mark.parametrize("host", ["0.0.0.0", "::", "[::]", "*", "", "192.168.1.10", "10.0.0.5"])
```

Ruff (0.16.1)

[error] Possible binding to all interfaces

(S104)